File: contivotos/lib/data/models/candidato.dart
class Candidato {
  final String id;
  final String nombre;
  final String propuesta;
  final String imagen;
  final String semestre;
  final String cargo;
  final int numero;
  final String experiencia;
  final String vision;

  Candidato({
    required this.id,
    required this.nombre,
    required this.propuesta,
    required this.imagen,
    required this.semestre,
    required this.cargo,
    required this.numero,
    required this.experiencia,
    required this.vision,
  });

  factory Candidato.fromFirestore(id, data) {
    return Candidato(
      id: id,
      nombre: data['nombre'] ?? '',
      propuesta: data['propuesta'] ?? '',
      imagen: data['imagen'] ?? '',
      semestre: data['semestre'] ?? '',
      cargo: data['cargo'] ?? '',
      numero: (data['numero'] ?? 0) as int,  // Asegurar que sea int
      experiencia: data['experiencia'] ?? '',
      vision: data['vision'] ?? '',
    );
  }
}

File: Backend/schemas/candidato.py
from pydantic import BaseModel, Field
from typing import Optional


# =========================
# Schemas Candidato
# =========================


class CandidatoBase(BaseModel):
    nombre: str
    numero: int
    cargo: Optional[str] = None
    imagen: Optional[str] = None
    propuesta: Optional[str] = None
    vision: Optional[str] = None
    experiencia: Optional[str] = None
    semestre: Optional[str] = None


class CandidatoCreate(CandidatoBase):
    """Schema para crear un nuevo candidato"""
    pass


class CandidatoUpdate(BaseModel):
    """Schema para actualizar un candidato (todos los campos opcionales)"""
    nombre: Optional[str] = None
    numero: Optional[int] = None
    cargo: Optional[str] = None
    imagen: Optional[str] = None
    propuesta: Optional[str] = None
    vision: Optional[str] = None
    experiencia: Optional[str] = None
    semestre: Optional[str] = None
    votos: Optional[int] = None


class Candidato(CandidatoBase):
    """Schema para respuesta con candidato completo"""
    id: str
    votos: int

    class Config:
        from_attributes = True
